Lib/imaplib.py

```diff
@@ -588,6 +648,48 @@ def getquotaroot(self, mailbox):
         return typ, [quotaroot, quota]
 
 
+    def idle(self, duration=None):
+        """Return an Idler: an iterable context manager for the IDLE command
+
+        :param duration:  Maximum duration (in seconds) to keep idling,
+                          or None for no time limit.
+                          To avoid inactivity timeouts on servers that impose
+                          them, callers are advised to keep this <= 29 minutes.
+                          See the note below regarding IMAP4_stream on Windows.
+        :type duration:   int|float|None
+
+        The context manager sends the IDLE command upon entry, produces
```

Comment:
I think the docstring should be smaller. We should rely on the online documentation rather than the docstring. But it's fine to include all those details in the .rst (let's avoid duplicating the docs since otherwise we need to sync them everytime).